File: src/Application Specific/Forms And Controls/Order/DataGridOrderTextBoxColumn.vb
Imports System.Windows.Forms.DataGridTextBoxColumn

Public Class DataGridOrderTextBoxColumn
    Inherits DataGridTextBoxColumn

    Public Sub New()
        MyBase.New()

        Me.TextBox.BackColor = Color.LightYellow
    End Sub


    Protected Overrides Sub Abort(ByVal rowNum As Integer)
        MyBase.Abort(rowNum)
    End Sub

    Protected Overrides Function Commit(ByVal dataSource As System.Windows.Forms.CurrencyManager, ByVal rowNum As Integer) As Boolean
        Return MyBase.Commit(dataSource, rowNum)
    End Function

    Protected Overloads Overrides Sub Edit(ByVal source As System.Windows.Forms.CurrencyManager, ByVal rowNum As Integer, ByVal bounds As System.Drawing.Rectangle, ByVal [readOnly] As Boolean, ByVal instantText As String, ByVal cellIsVisible As Boolean)
        Me.TextBox.Enabled = True
        MyBase.Edit(source, rowNum, bounds, False, instantText, cellIsVisible)
    End Sub

    Protected Overrides Function GetMinimumHeight() As Integer
        Return MyBase.GetMinimumHeight
    End Function

    Protected Overrides Function GetPreferredHeight(ByVal g As System.Drawing.Graphics, ByVal value As Object) As Integer
        Return MyBase.GetPreferredHeight(g, value)
    End Function

    Protected Overrides Function GetPreferredSize(ByVal g As System.Drawing.Graphics, ByVal value As Object) As System.Drawing.Size
        Return MyBase.GetPreferredSize(g, value)
    End Function

    Protected Overloads Overrides Sub Paint(ByVal g As System.Drawing.Graphics, ByVal bounds As System.Drawing.Rectangle, ByVal source As System.Windows.Forms.CurrencyManager, ByVal rowNum As Integer)
        MyBase.Paint(g, bounds, source, rowNum)

    End Sub

    Protected Overloads Overrides Sub Paint(ByVal g As System.Drawing.Graphics, ByVal bounds As System.Drawing.Rectangle, ByVal source As System.Windows.Forms.CurrencyManager, ByVal rowNum As Integer, ByVal alignToRight As Boolean)
        MyBase.Paint(g, bounds, source, rowNum, alignToRight)
    End Sub
End Class

File: apollo/apollo.fsm/src/Application Specific/Forms And Controls/Order/DataGridOrderTextBoxColumn.cs
using System.Drawing;
using System.Windows.Forms;

namespace FSM
{
    public class DataGridOrderTextBoxColumn : DataGridTextBoxColumn
    {
        public DataGridOrderTextBoxColumn() : base()
        {
            TextBox.BackColor = Color.LightYellow;
        }

        protected override void Abort(int rowNum)
        {
            base.Abort(rowNum);
        }

        protected override bool Commit(CurrencyManager dataSource, int rowNum)
        {
            return base.Commit(dataSource, rowNum);
        }

        protected override void Edit(CurrencyManager source, int rowNum, Rectangle bounds, bool readOnly, string instantText, bool cellIsVisible)
        {
            TextBox.Enabled = true;
            base.Edit(source, rowNum, bounds, false, instantText, cellIsVisible);
        }

        protected override int GetMinimumHeight()
        {
            return base.GetMinimumHeight();
        }

        protected override int GetPreferredHeight(Graphics g, object value)
        {
            return base.GetPreferredHeight(g, value);
        }

        protected override Size GetPreferredSize(Graphics g, object value)
        {
            return base.GetPreferredSize(g, value);
        }

        protected override void Paint(Graphics g, Rectangle bounds, CurrencyManager source, int rowNum)
        {
            base.Paint(g, bounds, source, rowNum);
        }

        protected override void Paint(Graphics g, Rectangle bounds, CurrencyManager source, int rowNum, bool alignToRight)
        {
            base.Paint(g, bounds, source, rowNum, alignToRight);
        }
    }
}